### PR TITLE
Update computation of proposal logp ratio in SMC IMH kernel

### PR DESCRIPTION
**Pull request by ricardoV94**
In a conversation with @aloctavodia we realized that the computation of the acceptance ratio was slightly off for the IMH kernel. @aloctavodia wrote down the equations on paper so we should be pretty confident this is correct :D

### CODE REVIEW

File: pymc/smc/smc.py [hunk 1]
```diff
@@ -379,31 +379,31 @@ def tune(self):
...
         """Independent Metropolis-Hastings perturbation."""
         ac_ = np.empty((self.n_steps, self.draws))
-
-        cov = self.proposal_dist.cov
         log_R = np.log(self.rng.random((self.n_steps, self.draws)))
+
+        # The proposal is independent from the current point.
+        # We have to take that into account to compute the Metropolis-Hastings acceptance
+        # We first compute the logp of proposing a transition to the current points.
+        backward_logp = self.proposal_dist.logpdf(self.tempered_posterior)
```
Comment (aloctavodia) [inline pymc/smc/smc.py:389]: This should be inside the loop, self.tempered_posterior is updated at each n_step

Comment (ricardoV94) [reply pymc/smc/smc.py:389]: I update this variable inplace at the end of the loop, with the logp of the accepted transitions

Comment (aloctavodia) [reply pymc/smc/smc.py:389]: Ohh I see now!

**Review by aloctavodia — APPROVED** [review]

File: pymc/smc/smc.py [hunk 2]
```diff
@@ -379,31 +379,31 @@ def tune(self):
...
-                (proposal_logp + backward) - (self.tempered_posterior_logp + forward)
+                (proposal_logp + backward_logp) - (self.tempered_posterior_logp + forward_logp)
             )
+
             ac_[n_step] = accepted
             self.tempered_posterior[accepted] = proposal[accepted]
             self.tempered_posterior_logp[accepted] = proposal_logp[accepted]
             self.prior_logp[accepted] = pl[accepted]
             self.likelihood_logp[accepted] = ll[accepted]
+            backward_logp[accepted] = forward_logp[accepted]
```
Comment (junpenglao) [inline pymc/smc/smc.py:408]: So this is the same as running `backward_logp = self.proposal_dist.logpdf(self.tempered_posterior)` here again right?

Comment (ricardoV94) [reply pymc/smc/smc.py:408]: Yes, just avoiding the recomputation because we already have all the values in memory

Comment (ricardoV94) [reply pymc/smc/smc.py:408]: Added a comment at the top clarifying this, as it was brought up twice already :)

**Pull request merged**